ga4gh/datamodel/reads.py

```diff
@@ -271,42 +316,12 @@ class HtslibReadGroup(datamodel.PysamDatamodelMixin, AbstractReadGroup):
     """
     A readgroup based on htslib's reading of a given file
     """
-    def __init__(self, parentContainer, localId, dataFile):
+    def __init__(self, parentContainer, localId):
         super(HtslibReadGroup, self).__init__(parentContainer, localId)
-        self._samFilePath = dataFile
-
-    def openFile(self, dataFile):
-        return pysam.AlignmentFile(dataFile)
-
-    def getSamFilePath(self):
-        """
-        Returns the file path of the sam file
-        """
-        return self._samFilePath
-
-    def getNumAlignedReads(self):
-        samFile = self.getFileHandle(self._samFilePath)
-        return samFile.mapped
-
-    def getNumUnalignedReads(self):
-        samFile = self.getFileHandle(self._samFilePath)
-        return samFile.unmapped
-
-    def getPrograms(self):
-        programs = []
-        samFile = self.getFileHandle(self._samFilePath)
-        if 'PG' not in samFile.header:
-            return programs
-        htslibPrograms = samFile.header['PG']
-        for htslibProgram in htslibPrograms:
-            program = protocol.Program()
-            program.id = htslibProgram['ID']
-            program.commandLine = htslibProgram.get('CL', None)
-            program.name = htslibProgram.get('PN', None)
-            program.prevProgramId = htslibProgram.get('PP', None)
-            program.version = htslibProgram.get('VN', None)
-            programs.append(program)
-        return programs
+        self._parentContainer = parentContainer
```

Comment:
This should already be set in the superclass, as we're passing it as a parameter to ``__init__``